packages/block-editor/src/components/iframe/index.js

```diff
@@ -306,16 +306,29 @@ function Iframe( {
 		iframeDocument.documentElement.classList.add( 'is-zoomed-out' );
 
 		const maxWidth = 750;
+		const frameSizeIsNumber = frameSize === 'number';
+		// The added space from `frameSize` has to be accounted for in scale calculation.
+		// This just punts in case `frameSize` can’t be treated as a pixel value. It’d
+		// probably be good to formally type `frameSize` as number and treat it as pixels.
+		// Core usage is pixels only and supporting any unit would be complicated.
+		let frameWidth = 0;
+		if ( frameSizeIsNumber || frameSize.endsWith( 'px' ) ) {
+			frameWidth =
```

Comment:
when are we not in any of these conditions?

Reply:
We, as in core, never. `frameSize` is a public prop and given that `VisualEditor` specifies it as as`'48px'` some third-party could conceivably have the notion that any CSS unit could be used and be passing `2em`. I revised the comment preceding this to hopefully clarify.

Reply:
The `IFrame` component is a public API, so it could have, in theory, already been used by a third-party with a non-pixel value. Making this a breaking change.

Reply:
Thanks for commenting. I think there is no breaking changes. Maybe that comment can still be better phrased. The `frameSize` never gets left out of the applied styles, it just doesn’t get factored into the "default" scale calculation when it can’t be treated as pixels. That’s how it was for any value before this.

I did, thanks to your comment, reexamine everything and realized the second commit here was preventing a non-pixel `frameSize` from being counter-scaled in the styles. So I pushed a 1374d0bda67e653240cea65466af038f7f64f187 to restore that. I’d be pleased to find a way to improve the poetry here and will look for a better refactor.

Reply:
> > The `IFrame` component is a public API, so it could have, in theory, already been used by a third-party with a non-pixel value. Making this a breaking change.
>
> I think this is not a breaking change. The `frameSize` never gets left out of the applied styles, it just doesn’t get factored into the "default" scale calculation when it can’t be treated as pixels. That’s how it was for any value before this.

I think I see what you're saying, but the effect for consumers that were using non-pixel units is still broken with this PR as it no longer adds horizontal padding for non-pixel units like trunk, thus the breaking change.

|      | Trunk | #65814 |
|------|-------|--------|
| 48px | ![Trunk 48px](https://github.com/user-attachments/assets/9c368057-b09d-42e4-b571-791f8ccce6e1) | ![PR 48px](https://github.com/user-attachments/assets/bd01a4ca-6a9e-46ea-a1a1-5fa17277361c) |
| 8em  | ![Trunk 8em](https://github.com/user-attachments/assets/d8f49939-a14b-479f-978a-c6cdd615998b) | ![PR 8em](https://github.com/user-attachments/assets/cf094ce5-5685-4187-93d3-acfc4c9d82a5) |

> I did, thanks to your comment, reexamine everything and realized the second commit here was preventing a non-pixel `frameSize` from being counter-scaled in the styles. So I pushed a [1374d0b](https://github.com/WordPress/gutenberg/commit/1374d0bda67e653240cea65466af038f7f64f187) to restore that. I’d be pleased to find a way to improve the poetry here and will look for a better refactor.

My comment was mostly just a gut feeling from having read the code which is why it didn't have that much context. You had me reexamine my assumptions too, so I do appreciate that.

I figured we could move your `frameWidth` calculation to CSS to get around the issue, so I dug into that and found out that CSS could not actually handle the math for units; my assumption was wrong.

The division operator in CSS `calc()` doesn't do the dimensional analysis of first converting units to pixels and then getting the ratio of them.

I've been wracking my brain to think of ways to support units so that it isn't a breaking change because including the `frameSize` is important for calculating the scale.

I suppose a loophole for the backwards compatibility argument could be that nothing about the prop was documented, so you could argue that the previous API accepted only numbers or pixel strings rather than any CSS unit. I think I'm okay with that as long as we add documentation for the `px` requirement now even if it will break non-px usage that may exist.